spec.bs

```diff
@@ -2444,6 +2448,27 @@ algorithms to achieve the outcomes described in the above explanatory content.
   /fenced-frame/permission-notification.https.html
 </wpt>
 
+<h3 id=cssom-monkeypatch>CSSOM View</h3>
+
+The [[!CSSOM-VIEW]] specification defines the {{Element/scrollIntoView()}} method that calls the
+[=scroll a target into view=] algorithm. This will not only scroll the {{Element}} or
+[=css2/viewport=] to make the target visible, but will also scroll [=tree/ancestor=]s if necessary
+to make the target visible, essentially causing the scroll to "bubble up". This means that
+{{Element/scrollIntoView()}} performed in a [=child navigable=] or [=fenced navigable
+container/fenced navigable=] can be observed by its embedder, allowing for collusion across a fenced
+frame boundary. This section patches {{Element/scrollIntoView()}} to prevent that collusion at the
+expense of some functionality.
+
+<div algorithm=scroll-target-into-view>
+  Modify the [=scroll a target into view=] algorithm to add a step at the end of the algorithm that reads:
+
+  14. If <var ignore>scrolling box</var>'s associated {{Element}} or [=css2/viewport=]'s associated
```

Comment:
Do we need to do anything about step 13 of this algorithm, which seems to unconditionally crawl up to the "outer" (at least I think that's what it's doing) viewport and call "perform the scroll" again, recursively? *That* seems to be the source of the recursion, no?

Reply:
The recursive part is the part at the start of the algorithm that says "means to run these steps for each ancestor element or viewport that establishes a scrolling box scrolling box".

Step 13 is the step that causes the element or viewport to do the actual scroll. Nothing up until that point actually does any scrolling (step 11 is misleading because it mentions aligning, but everything resolves to the `position` variable).

I also don't see any recursive calls in the steps themselves. Step 13 is calling the [perform a scroll](https://drafts.csswg.org/cssom-view-1/#perform-a-scroll) algorithm (which is [the algorithm also called](https://drafts.csswg.org/cssom-view-1/#ref-for-perform-a-scroll) when [a user scrolls with a trackpad or a touch drag](https://drafts.csswg.org/cssom-view-1/#example-vvanimation)), which is separate from the [scroll a target into view](https://drafts.csswg.org/cssom-view-1/#scroll-a-target-into-view) algorithm.

Because of that, I think what we currently have is correct.